python/cuda_parallel/cuda/parallel/experimental/_cccl_interop.py

```diff
@@ -178,20 +183,77 @@ def to_cccl_value(array_or_struct: np.ndarray | GpuStruct) -> Value:
         return to_cccl_value(array_or_struct._data)
 
 
-def to_cccl_op(op: Callable | None, sig) -> Op:
-    if op is None:
-        return Op(
-            operator_type=OpKind.STATELESS,
-            name=None,
-            ltoir=None,
-            state_alignment=1,
-            state=None,
-        )
+def _create_void_ptr_wrapper(op, sig):
+    """Creates a wrapper function that takes all void* arguments and calls the original operator.
 
-    ltoir, _ = cuda.compile(op, sig=sig, output="ltoir")
+    The wrapper takes N+1 arguments where N is the number of input arguments to `op`, the last
+    argument is a pointer to the result.
+    """
+    # Generate argument names for both inputs and output
+    input_args = [f"arg_{i}" for i in range(len(sig.args))]
+    all_args = input_args + ["ret"]  # ret is the output pointer
+    void_sig = types.void(*(types.voidptr for _ in all_args))
+
+    # Create the wrapper function source code
+    wrapper_src = textwrap.dedent(f"""
+    @intrinsic
+    def impl(typingctx, {', '.join(all_args)}):
+        def codegen(context, builder, impl_sig, args):
+            # Get LLVM types for all arguments
+            arg_types = [context.get_value_type(t) for t in sig.args]
+            ret_type = context.get_value_type(sig.return_type)
+
+            # Bitcast from void* to the appropriate pointer types
+            input_ptrs = [builder.bitcast(p, t.as_pointer()) for p, t in zip(args[:-1], arg_types)]
+            ret_ptr = builder.bitcast(args[-1], ret_type.as_pointer())
+
+            # Load input values from pointers
+            input_vals = [builder.load(p) for p in input_ptrs]
+
+            # Call the original operator
+            result = context.compile_internal(builder, op, sig, input_vals)
+
+            # Store the result
+            builder.store(result, ret_ptr)
+
+            return context.get_dummy_value()
+        return void_sig, codegen
+
+    # intrinsics cannot directly be compiled by numba, so we make a trivial wrapper:
+    def wrapped_{op.__name__}({', '.join(all_args)}):
+        return impl({', '.join(all_args)})
+    """)
```

Comment:
Let's save repeatedly used `', '.join(all_args)` into a variable before this template and use the variable in the template to avoid re-computing string of joined arguments 3 times.